shell/browser/serial/serial_chooser_context.cc

```diff
@@ -0,0 +1,76 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "shell/browser/serial/serial_chooser_context.h"
+
+#include <utility>
+
+#include "base/base64.h"
+#include "base/memory/singleton.h"
+#include "content/public/browser/device_service.h"
+#include "mojo/public/cpp/bindings/pending_remote.h"
+
+namespace electron {
+
+SerialChooserContext* SerialChooserContext::GetInstance() {
+  return base::Singleton<SerialChooserContext>::get();
```

Comment:
`base::Singleton`'s documentation says to avoid using it if possible.

Can we tie the lifetime of this to something that already has a well-defined lifetime? (e.g. `ElectronBrowserContext`)?

If we don't need to invoke the destructor, can we use `base::NoDestructor` instead?

Reply:
The chromium implementation uses a `BrowserContextKeyedServiceFactory`, so I changed our implementation to do the same.